### DISCUSSION TITLE
Support for more than one Zigbee coordinator (and networks) in ZHA at the same time?

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#673] [ZHA development]
Will you please consider adding support for more than one Zigbee coordinator in the ZHA integration for the same Home Assistant instance?

Limitation: Users can only add one ZHA integration to Home Assistant (but HA can connect to multiple deCONZ or Zigbee2MQTT instances).

Requesting that one single instance of the ZHA integration in Home Assistant could be extended be able to support multiple Zigbee coordinator adapters/dongles/sticks, with each Zigbee coordinator adapter/dongle/stick, having their own separate Zigbee network mesh (which have no knowledge of any other Zigbee coordinators or their Zigbee network meshes that is being used by that same ZHA instance). 

ZHA support for multiple Zigbee USB sticks/dongles is a feature that has been requested many times in Home Assistant community forums:
* https://community.home-assistant.io/t/zha-multiple-co-ordinators/255101
* https://community.home-assistant.io/t/zha-multiple-gateway/262101/
* https://community.home-assistant.io/t/zha-support-for-more-than-one-zigbee-coordinator-in-zha/258248/
* https://community.home-assistant.io/t/zigbee-zha-are-multiple-coordinators-possible/224050
* https://community.home-assistant.io/t/multiple-zha-integrations-possible/478588/
* https://community.home-assistant.io/t/multiple-sonoff-zigbee-bridge/264491
* https://community.home-assistant.io/t/multiple-zigbee-networks-radios-on-one-hub/160056/

Examples of possible use-case scenarios: 

- Increase coverage if the user has a separate remote detached building like example a garden-shed that for whatever can not reach via your primary Zigbee network mesh via it RF radio signals but that remote building does have wired Ethernet on the same LAN as your Home Assistant installation, in which case the solution could be for the user to buy and use a network-attached Zigbee coordinator bridge as their second Zigbee coordinator adapter/dongle to that same ZHA integration instance.

- Ability to have one "production" Zigbee network and one "test" Zigbee network in the same ZHA instance. That way you can safely test new Zigbee devices and/or isolate unstable Zigbee devices to troubleshoot on a second network so as not to affect the stability of your "production" Zigbee network, (which might be something that could make your wife/girlfriend/partner/roommate less irritated about you testing and troubleshooting Zigbee devices as well).

- Ability to run two Zigbee coordinators from different manufacturers/radio-libraries at the same time in one and the same ZHA instance.

If implemented it would be nice if could select which specific Zigbee coordinator stick/dongle that you want to enable pairing on.

Note! This request is as such made with the full understanding that each Zigbee coordinator will have their own separate Trust Center and only ever control its own separate Zigbee network. But since the Zigbee specification does not support multiple coordinators on the same network it would be nice if there could be a workaround implemented at the ZHA application level to allow several Zigbee coordinators as long as they have their own separate networks that are not aware of each other at the Zigbee level.

Another alternative could be to make it easier to run multiple separate instances of ZHA in one and the same Home Assistant instance(?).

PS: FYI, OpenHAB Zigbee binding addon/integration does support multiple Zigbee coordinator adapters/dongles/sticks, with each Zigbee coordinator adapter/dongle/stick, having their own separate Zigbee network mesh, see https://www.openhab.org/addons/bindings/zigbee/

PPS: Z-Wave solves this whole problem by allowing multiple controllers on the same network, (only one primary Z-Wave but you can have several secondary Z-Wave controllers in the same network).

**Comment by MattWestb** [comment]
Perhaps its possible implanting one fullfutured coordinator with trustcenter and and pointing the "satellites" to using the masters trustcenter. Back draw is that the the trustcenter is down all network is degraded (atleast turdtcenter service is not working but ZLL devices is still working.

**Comment by Adminiuga** [comment]
can you please stop spamming with the requests??? We've heard it loud and clear in the past and nothing has changed since the last similar requests. Instead of the requests, better submit PRs.

**Reply by Hedda** [reply]
Previous request was asking for this feature to be implemented into zigpy, while this new request is now for ZHA/HA.

Reason for me posting idea suggestion as request and not PRs is that think my own coding skills are not good enough.

But if you really still feel that posting such suggestion ideas are "spam" then should I stop posting feature requests?

**Comment by Hedda** [comment]
Continuing discussion from https://github.com/zigpy/zigpy/discussions/635 about extending control coverage to different areas where Zigbee wireless protocol cannot reach: 

MattWestb wrote:
> Zigbee is allowing more controlling devices in one network !!
> But it can only being one trust center if you is reading the speces.
> And you can have "Zigbee control bridges" (that is wot IKEA GW is) (Control bridge [Device ID: 0x0840]) that is controlling other devices in the network as it was one own network(or part of the network as private ones).

Are you saying a single Zigbee network can have a separate "Zigbee Control Bridge" in different locations including separate buildings in RF dead zones not reachable via Zigbee wireless signals as long those Zigbee Control Bridges are connected to same LAN (Ethernet or WiFi)? 

Meaning that the communication between "Zigbee Control Bridge" to  "Zigbee Control Bridge" does not have to go over Zigbee wireless signals as a medium? If so then that concept sounds very similar to multi-controller support in Z-Wave specification. 

What would it take for zigpy + radio libraries and ZHA to add support to such Zigbee Control Bridge specification(s) and support?

**Reply by MattWestb** [reply]
Nop the controller bridge is only sending and receiving commands to / from the network is associated to with CLI or application interface API support. You can see it as one "flexible router" that can act as only one router without function or more advanced like IKEA GW and most other ZLL controllers (Scene controllers and GW / bridges with no TC).
The question was "allowing multiple controllers on the same network" and yes more controllers is possible but only one trust center if using centralized security and the trust center is not one controller its the security manager part in one centralized zigbee network. 
Controller bridge is only part of the network and can using all functions in the network as long not some device have requesting on private APS key from the coordinator and is using one "private encrypted pipe" between 2 or more devices that the controller bridge not have getting then its "private" for the devices that have requesting it. All other functions the controller bridge can accessing and manipulating.

The spec for [ZigBee Lighting & Occupancy Device Specification](https://zigbeealliance.org/wp-content/uploads/2019/11/docs-15-0014-05-0plo-Lighting-OccupancyDevice-Specification-V1.0.pdf) "25 Control bridge".

But its being little more then you is saying "only one coordinator".
You can having your trust center in the coordinator and is pointing to the address 0x0000.
But you can have the TC on one other address = distributed and pointing to the TC you is  using in your (zigbee) network.

I cant saying how its working but you need digging very deep i the information for finding how its working.

Perhaps the IP  implanting of Zigbee can bridging between zigbee pools of devices that dont have direct radio contact but i have not looking on that.

**Reply by Hedda** [reply]
> The question was "allowing multiple controllers on the same network"

That was not my question as I only mentioned Z-Wave supports that. Question was for coordinators with separate networks:

> This is why I like to see support for more than one Zigbee coordinator (and networks) in ZHA at the same time, as requested here:
> 
> #673
> 
> **Having support for two Zigbee coordinators in ZHA (with their own separate mesh networks)** would solve this problem you get when you have a single property with two areas that can not for whatever signal killing reason be bridged using one mesh network.
> 
> Z-Wave solves this by allowing multiple controllers on the same network, (only one primary Z-Wave but you can have several secondary Z-Wave controllers in the same network).
> 
> **Since the Zigbee specification does not have a similar solution it would be nice if there could be a workaround implemented at the ZHA application level to allow several Zigbee coordinators as long as they have their own separate networks that are not aware of each other at the Zigbee level.**

**Reply by Hedda** [reply]
> Perhaps the IP implanting of Zigbee can bridging between zigbee pools of devices that dont have direct radio contact but i have not looking on that.

If thinking of the [Connected Home over IP (CHIP) project](https://www.connectedhomeip.com/) then know that Zigbee is sadly not in-scope for their first version.

That is, if you read more about that project you find that the companies behind the Zigbee Alliance is collaborating on that project but the actual adding of IPv6 support over Zigbee is not currently their primary goal. Right now they as a proof-of-concept currently only looking at adding support for Thread, BLE, and WiFi (ESP32). CHIP over Zigbee is today only a potential candidate that might get added to their roadmap in the future.

https://github.com/project-chip/connectedhomeip -> https://github.com/project-chip/connectedhomeip/projects

PS: IP is also a concept where Z-Wave specifications was ahead with "Z/IP" (Z-Wave over IP a.k.a. ZIP) gateway + router:

https://www.silabs.com/developers/z-wave-controller

**Key Benefits of Z-IP Gateway Features**
- IP applications reach Z-Wave nodes via IP addresses
  - Z/IP Packets from LAN IP applications are terminated and forwarded as Z-Wave commands to Z-Wave Nodes
  - Z-Wave nodes are identified by IPv6 and IPv4 host addresses
  - Resource & Service Discovery (mDNS)
  - Remote Access and Configuration
- Z-Wave SmartStart Auto-inclusion
- Z-Wave Security 2 support
- Z-Wave Mailbox Service – Support for Non-Listening nodes
- Installation and Maintenance framework
- Firmware Update of Z-Wave module attached to the Z/IP Gateway
- Migration to Z/IP Gateway from non-Z/IP Gateways
- Back-up/Recover of Z-Wave network and product data

![image](https://user-images.githubusercontent.com/6320001/112617426-42c3e280-8e25-11eb-892a-3890aad31928.png)

**Comment by pipiche38** [comment]
My humble opinion is that it is not taken in consideration from the Zigbee alliance and so you cannot make it.
Now practically, the way I'm doing is not to use 1 Zigbee controller for my home as I have indeed a Single Point of Failure which is the controller. To mitigate that risk (as we cannot avoid it), I prefer to have a Controller per Zone and all controllers are managed directly by the Application I'm using ( Domoticz ) where I can run several instances of the plugin and where my scenarios are there.
So in case of a controller failure, then the others are still alive .
In short I'm not fan at all of the 100 more controller which indeed increase the risk level

**Reply by MattWestb** [reply]
@pipiche38 Is you have centralized (TC) or distributed security in your Zigbee networks ?
Distributed is making standalone autonomous network parts with the same network parameters but have address and group rage the controller can controlling (like IKEA GW is doing).

Im not one fan of single point of failure things like TC or automatons in node-red that only working if all systems i online.

The Zigbee Light link is one great future for getting the lights working also then the host system or coordinator is off line.

By the way you can adding some more tuya TRVs ID that is being "found" that is matching that you have implanted (from the last time i was looking in your repro) :-)))

**Comment by Hedda** [comment]
> Requesting that one single instance of the ZHA integration in Home Assistant could be extended be able to support multiple Zigbee coordinator adapters/dongles/sticks, with each Zigbee coordinator adapter/dongle/stick, having their own separate Zigbee network mesh (which have no knowledge of any other Zigbee coordinators or their Zigbee network meshes that is being used by that same ZHA instance).

FYI, it sounds as like the new Jeedom Zigbee Plugin has implemented this feature (translated from French to English with Google):

https://blog.jeedom.com/5183-tout-ce-quil-faut-savoir-sur-le-plugin-officiel-zigbee/

"**_The Zigbee plugin is also capable of managing several controllers at the same time and therefore several Zigbee networks within the same Jeedom instance. 2 separate Zigbee networks are currently supported but this number could very simply be increased if the demand arises._**"

**Reply by MattWestb** [reply]
I think i need learning french ;-(

I have only learned using "La vache" and its little danger saying to one woman in the supermarket !!

**Reply by Hedda** [reply]
Google translate is your friend:

https://translate.google.com/translate?sl=fr&tl=en&u=https://blog.jeedom.com/5183-tout-ce-quil-faut-savoir-sur-le-plugin-officiel-zigbee/

**Comment by pipiche38** [comment]
If I understood the Jeedom plugin is based on the zigpy work. Is my understand wrong that as zigpy is under the GNU license, that the Jeedom plugin must also "By contrast, the GNU General Public License is intended to guarantee your freedom to
share and change all versions of a program--to make sure it remains free software for all its users."

From my understanding the Jeedom plugin is a paid plugin and the source code seems not available as well.

Is that not violating the GNU license ?

**Reply by MattWestb** [reply]
I think (=not knowing) that you is right about publicizing the the source code is one must.

For the moment is Jeedom helping the Zibpy project moving forwards with fundamental implanting functions in the core code that is good for both projects, but in the long run its can being problems if not publishing the used source code they is using and that more of one "political" problem that must being solved between the both projects in the end.

And sorry for may bad french but im always on the hunt for [Lavash bread](https://en.wikipedia.org/wiki/Lavash) also then shopping in at Carrefour and i cant pronouncing it so well :-((

And im more sorry for that Domoticz was not getting the SQLITE problems solved so can using Zigpy as one add on and getting support for more hardware in the system that i think have being one win  win situation for both projects !!

**Reply by MattWestb** [reply]
By the way its looks its start moving on the "other side" https://github.com/domoticz/domoticz/commit/b34627e48eb2fcf96b5ce713741a638740d5d4eb so perhaps it can being one way moving forwards in the future ?

**Reply by Hedda** [reply]
> If I understood the Jeedom plugin is based on the zigpy work. Is my understand wrong that as zigpy is under the GNU license, that the Jeedom plugin must also "By contrast, the GNU General Public License is intended to guarantee your freedom to
> share and change all versions of a program--to make sure it remains free software for all its users."
> 
> From my understanding the Jeedom plugin is a paid plugin and the source code seems not available as well.
> 
> Is that not violating the GNU license ?

I believe that the paying for plugins/addons via a portal (e.g a digital app store/marketplace) part is OK as you can then see it as you are then paying for the service of automatic updates etc. via the plugin/addon portal which needs a maintained infrastructure on the backend side, (similar to how you can pay for the service to get the source code sent to you on a CD-R/DVD-R via snailmail), but whether or not the source code for the plugin needs to be fully made open source or not depends on whether or not the plugin only uses open APIs/CLIs to communicate with zigpy and the radio libraries or if the integration goes deeper.  

My guess that this is all just a miss by their developers that the source code is not yet available (i. e. a only the private repo that has not yet made public on GitHub because they simply forgot). Both the Jeedom core and other first-party plugins from the same developers have their source code already available, and I think it is probably OK to at least give them a small grace period before start nagging them about it. That is, my guess is that they will make the source code for the code using zigpy available sooner rather than later.

https://github.com/jeedom

https://github.com/jeedom/core

https://github.com/jeedom/plugin-openzwave

@zoic21 Any comments on your plans for source code availability of your new Zigbee plugin that depends on zigpy?

Will it soon be available at https://github.com/jeedom/plugin-zigbee or similar repo and URL as a public repository?

**Reply by Hedda** [reply]
> more sorry for that Domoticz was not getting the SQLITE problems solved so can using Zigpy as one add on and getting support for more hardware in the system that i think have being one win win situation for both projects

@pipiche38 Was the root cause for that not resolved with https://github.com/domoticz/domoticz/commit/b34627e48eb2fcf96b5ce713741a638740d5d4eb / https://github.com/domoticz/domoticz/commit/0a4ccfb82a7b2f291c821a9ac71b3d8504e125b7 ?

https://github.com/pipiche38/domoticz-sqlite3-issue closed https://github.com/pipiche38/domoticz-sqlite3-issue/issues/1 ?

**Comment by zoic21** [comment]
It's not really an oversight. Jeedom is under GNU GPLv3 like Zipgy. This license does not prohibit making access to the source chargeable. This is the case with Jeedom when you buy the plugin you automatically have access to the sources (they are on your jeedom). Anyway, the jeedom plugin does not include zigpy in these sources, it is installed afterwards on the Jeedom.

The case of access to paid sources is dealt with in the frequently asked questions of the site on the GPL:

-  https://www.gnu.org/licenses/gpl-faq.fr.html#DoesTheGPLAllowMoney
- https://www.gnu.org/licenses/gpl-faq.fr.html#CompanyGPLCostsMoney

**Comment by Hedda** [comment]
> This license does not prohibit making access to the source chargeable. This is the case with Jeedom when you buy the plugin you automatically have access to the sources (they are on your jeedom).

@zoic21 What it says is that if you buy the plugin yourself then you have the right to demand that you provide the source code for it, or?

https://www.gnu.org/licenses/gpl-faq.html#DoesTheGPLAllowMoney

**Q:** _Does the GPL allow me to sell copies of the program for money? (#DoesTheGPLAllowMoney)_
**A:** _Yes, the GPL allows everyone to do this. The right to sell copies is part of the definition of free software. Except in one special situation, there is no limit on what price you can charge. (**The one exception is the required written offer to provide source code that must accompany binary-only release.**)_

As you can read there you must provide a written offer to provide source code that must accompany binary-only release (i.e. the plugin).

> Jeedom is under GNU GPLv3 like Zipgy.

zigpy libraries are licensed as GNU GPLv3, but Jeedom core currently actually looks to be licensed under GNU GPLv2, not GNU GPLv3:

https://github.com/jeedom/core/blob/alpha/LICENSE

But what is your new official Zigbee plugin for Jeedom licensed under? Is its license compatible with GNU GPLv2 and/or GNU GPLv3?

I am not a license expert but read that if you link to a GPL library then you have created a derived work and your code must also be licensed under a GPL compliant license and its source code made available if you are given access to its binaries. Understanding that GPL is different to LGPL code which specifically allows dynamic linking of differently licensed code. GPLv2/GPLv3 licenses as such does not allow linking from differently licensed code unless it licensed under a compatible open source license. Is this not a correct understanding?

How is the code integration to zigpy done exactly? How is code integration to Jeedom done exactly? Plugins are not stand-alone as can not be used without Jeedom core and does not work without zigpy, so your plugin must at least be GNU GPLv2 and GNU GPLv3 compatible?

**Reply by Hedda** [reply]
> What it says is that if you buy the plugin yourself then you have the right to demand that you provide the source code

While you can legally do so I personally do not see the point in not providing open source code publicly online for collaboration purposes as anyone who buys the plugin can just turn around and upload the code to a public repo (like here on GitHub).

At least any GNU GPLv2 and/or GNU GPLv3 compatible license should allow user who buys the plugin to do so.

IMHO the only thing not providing the code publicly to other developers who depend on zigpy will just hinder collaboration. I mean, the only reason why Jeedom can even benefit from zigpy today is that Home Assistant developers made it available for ZHA. So would it not be fair to let future new project benefit from the seeing how Jeedom made their Zigbee implementation using zigpy?

You can still change for the plugin via the Jeedom market, (but again that is more like charging for installation service).

**Comment by zoic21** [comment]
I'm not good enough in licensing to answer ... I'm not even a Jeedom employee, just a volunteer who develops outside the working hours of his real job.

All I can say is that all the additions that Jeedom SAS makes on zigpy are offered in PR form as we do with Openzwave or any other library that Jeedom uses.

And as said before, the plugin absolutely does not embed zigpy or any other module in its code, it can only be accessed through the API provided by the zigpy library.

**Reply by pipiche38** [reply]
That is fair. Are you also relying on the zha-quirk to support the various devices , or do you developed your own ?
For instance Jeedom is communicating on the fact that they have with Danfoss integrated to their plugin the Danfoss Aly system, but as I have seen nothing as been pushed to zha-quirk

**Reply by zoic21** [reply]
We make a PR for the valve. For the main thermostat we open an issue to zigpy due to it's particularly. Thermostat use one endpoint for each control valve so we can not make a quirks...

**Comment by Hedda** [comment]
I understand that the rules become much clear cut if the question simply is only about not wanted to release the plugin source code at all; as then the answer is that while linking to LGPL libraries from closed programs are OK, linking to GPL libraries from closed programs is not OK. 

https://www.gnu.org/licenses/gpl-faq.en.html#GPLPlugins

https://www.gnu.org/licenses/gpl-faq.en.html#GPLAndPlugins

https://www.gnu.org/licenses/gpl-faq.en.html#GPLPluginsInNF

https://www.gnu.org/licenses/gpl-faq.en.html#NFUseGPLPlugins

https://www.gnu.org/licenses/gpl-faq.en.html#LinkingWithGPL

https://www.gnu.org/licenses/gpl-faq.en.html#SwitchToLGPL

**Reply by Hedda** [reply]
By the way, GPL questions should not only be about how the plugin uses zigpy but also how Jeedom core uses the plugin.

https://www.gnu.org/licenses/gpl-faq.en.html#GPLPlugins

**Comment by reesericci** [comment]
This jeedom stuff is a waste of time, let's try to get this implemented into zha/zigpy.

**Comment by mercenaruss** [comment]
Recently we was able to make it work properly.
Numerous ZigBee Coordinators over LAN, and numerous Z2M instances.
Also you can have 1 USB Zigbee Dongle and many LAN Coordinators.

Check this video:

https://www.youtube.com/watch?v=KBEaVZyvSgY&ab_channel=%D0%A3%D0%BC%D0%BD%D1%8B%D0%B9%D0%B4%D0%BE%D0%BC%D1%81AlexKvazis

**Comment by Hedda** [comment]
With the work on zha-websocket-server and zhaws-addon do you think Home Assistant could get support for multiple instances of ZHA?

https://github.com/zigpy/zha-websocket-server

https://github.com/zigpy/zhaws-addon

That is, run more than instances of ZHA in Home Assistant at same time (if can not than one Zigbee coordinator in ZHA at the same time)?

Or could the zha ws-addon be made to support serveral zha websocket servers?

**Comment by mccasian** [comment]
alternatively, you can just run multiple zigbee2mqtt instances instead running two separate HA instances. I've managed to connect 2 Bridges on my HA in a pretty easy way. 

See here: https://community.home-assistant.io/t/zigbee-zha-are-multiple-coordinators-possible/224050/11?u=mccasian

**Reply by Hedda** [reply]
> alternatively, you can just run multiple zigbee2mqtt instances instead running two separate HA instances. 

@mccasian Yes but that is very off-topic and only a workaround for end-users to use Zigbee2MQTT instead of Home Assistant's own [ZHA integration](https://www.home-assistant.io/integrations/zha) and as such have nothing to do with this discussion in the [zigpy](https://github.com/zigpy/zigpy) repository and a discussion/request about Home Assistant's ZHA integration. Please feel free to suggest using Zigbee2MQTT as a workaround for end-users asking about using multiple Zigbee Coordinator adapters in the [Home Assistant Community Forum](https://community.home-assistant.io/) but please understand and respect that discussion about using Zigbee2MQTT instead of Home Assistant's own [ZHA integration](https://www.home-assistant.io/integrations/zha) and [zigpy](https://github.com/zigpy/zigpy) does not belong here.

**Reply by pipiche38** [reply]
I'm not a HA user, but why HA is not allowing multiple instance of ZHA , that would ease. But agree this is not the subject here

**Reply by Hedda** [reply]
> I'm not a HA user, but **why HA is not allowing multiple instance of ZHA**, that would ease.

Users should be able to do so too today as a workaround in order to have multiple instasses of the ZHA integration component, but it would be a hacky solution as users would need to manually copy the [zha component](https://github.com/home-assistant/core/tree/dev/homeassistant/components/zha) to a new folder/directory and edit the [manifest JSON file](https://github.com/home-assistant/core/blob/dev/homeassistant/components/zha/manifest.json) on the second copy so that both its domain and name are named something else which are unique, which in practice would make it into a completely separate integration component, (though I believe doing so might mess up the USB discovery and Zeroconf discovery functions for initiation the config flow).

**Reply by mccasian** [reply]
>
>"**VERY** off-topic"

Oh, that really triggered you.

> 
> respect that discussion about using Zigbee2MQTT instead of Home Assistant's own [ZHA integration](https://www.home-assistant.io/integrations/zha) and [zigpy](https://github.com/zigpy/zigpy) does not belong here.

Since this thread provided no solution yet, I was just trying to say "hey, meantime you may try another way... this ZHA feature may not come soon enough".

 It's called "contribution to community" and definitely not harming the thread "respect". - I still don't get how you end up seeing it that way...

**Reply by pdecat** [reply]
Well, there is a solution with ZHA which I've been using for a few years now, but you made me figure this was not mentioned here:

use one or more instances of Home Assistant with the https://github.com/custom-components/remote_homeassistant component, and configure ZHA instances on them.

**Reply by Hedda** [reply]
@mccasian, I am sorry, apologize if you find my reply offensive I really meant no offence. I just wanted to quickly stop the off-topic discussion about Zigbee2MQTT as I do not want you or anyone else to continue with off-topic discussions about Zigbee2MQTT in this specific thread, hence I only tried to highlight that the suggested workaround to use Zigbee2MQTT that you posted in this thread is definitely off-topic here, and I do still stand by that it is very off-topic considering my original post and this being the zigpy repository that you posted it. This thread is specifically meant to be a feature request directed towards zigpy/ZHA developers (as per its topic, original post and fact that was posted under the category "[ZHA development](https://github.com/zigpy/zigpy/discussions/categories/zha-development)") and not a general support question directed towards the Home Assistant community asking for alternative solutions. As this thread is not about workarounds your suggestion to use Zigbee2MQTT instead zigpy/ZHA does not contribute to this specific thread. Discussing non-zigpy/ZHA workaround is better suited to Home Assistant Community Forum -> https://community.home-assistant.io/ (there I too many times suggested using Zigbee2MQTT as a workaround until zigpy/ZHA gets support to use multiple Zigbee Coordinator adapters and Zigbee networks at the same time).

**Reply by mccasian** [reply]
it might be that I also got a little triggered :) I think people have different mindsets, and mine was that not only the author, but some other "working solution seekers" will read this thread... some will be happy with the alternatives.

I agree though, that the ZHA with multiple coordinators support in a single HA instance is THE SOLUTION, so thanks for bringing that need to the light.

**Reply by MattWestb** [reply]
Perhaps ZHA is not the solution then Matter is implanted and its using ZCL R8 + and Zigbee R23 i think ZHA can only piping the Zigbee trafic to the Matter instance and matter is doing the rest in HA or its opening for implanting multiple ZHA instances.

As @pdecat was saying `remote_homeassistant` looks being one great thing but i have not getting ZHA devices connected and piped to my production system from my 5 test rigs but i must digging more and see if its working.

**Reply by MattWestb** [reply]
As @pdecat was saying `remote_homeassistant` is working but the instruction is "little" out of date so its little try and error.
Good things you is getting all `entities` from the remote systems but no device objects and cant editing them in the master system (read only) and therefor also they is not in one aria so not easy organizing then was getting over 170 new entities.

Events can being subscribed like `zha_event` but link to the device / entities is broken the looking in the master log.

I think `remote_homeassistant` is one god way extending one HA system with more functionality and not putting all in one hardware and having more local systems collecting data and forwarding it to the master system.

**Comment by MattWestb** [comment]
Im waiting for Thread with backbone and multiple 801.15.4 network with boarder routers and redundant mesh network and host system.
